### DISCUSSION TITLE
[2023-09-25] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#7886] [Weekly Demo Day]
- James: S3 Read Access, Input Stream based reading. #7776
- James: Using conversions in a few places to improve Table APIs and JSON parsing. #7859
- James: Design work and prototyping on the Enso connectors from the VDD.
- Radosław: More robust Date/Time format patterns parsing #7826
- Radosław: Follow up improvements to `Date_Time_Formatter` #7875
- Radosław: First steps in refactoring Table problem handling #7879
- Greg: New Data.post for sending a payload to a Web API #7700
- Greg: Improve HTTP tests #7847
- Dmitry: Deliver information about `Xyz.from` conversion methods to the IDE #7716
- Jaroslav: Avoid StackOverflow when comparing unknown foreign objects #7780
- Jaroslav: Instrumenter to observe behavior of nodes with UUID #7833
- Jaroslav: Encapsulate TruffleLanguage.Env in EnsoContext #7842
- Jaroslav: Accept type argument in instance method signatures #7848
- Jaroslav: Avoid duplicated conversion & search target type scope #7849
- Jaroslav: Adjusting to GraalVM for JDK21+ #7855
- Jaroslav: Upgrading to Frgaal compiler 20.0.1 #7860
- Jaroslav: Downloadable VSCode extension #7861
- Jaroslav: Meta.Type.find a type by FQN #7885
- Hubert: Improve shutdown logic of language server #7801
- Hubert: Rename Decimal to Float #7807
- Hubert: Reduce extra output in compilation and tests #7809
- Hubert: Always log to console and file #7825
- Hubert + Pavel: Small wins in Engine CI job execution #7846
- Pavel: Do not suppress panic exception in runtime type check #7793
- Pavel: Implement private modules #7840
- Pavel: Run Enso tests with assertions enabled #7882
- Pavel: Implement Enso-specific assert #7883
- Dmitry: Log visualization expressions generated by IDE #7756
- Dmitry: Download single edition by name #7771
- Dmitry: Static parameter in search/completion request affects only methods #7831
- Dmitry: Send conversion method suggestions #7857
- Dmitry: Fix API representation of conversion suggestion #7862
- Dmitry: Open projects using the packaged language server #7868
- Cassandra: Set up laptop, got build working, completed several bookclubs, fraud demo use case design and review

**Comment by radeusgd** [comment]
ISO weekyear parsing:
![image](https://github.com/enso-org/enso/assets/1436948/00b42563-0a26-4bb8-a301-ac6370241769)

Various warnings:
![image](https://github.com/enso-org/enso/assets/1436948/21e59d01-1325-4cbe-b24a-db9930e51920)

Better error messages:
![image](https://github.com/enso-org/enso/assets/1436948/518a2450-2fe2-4080-99c5-6a2ea09f2d0d)
![image](https://github.com/enso-org/enso/assets/1436948/92a21773-4b05-498b-a201-9604df4ff530)

**Comment by jdunkerley** [comment]
Table Conversions:
![image](https://github.com/enso-org/enso/assets/4699705/99e4d579-bdad-4ad9-b937-53f5749a917d)

JSON Parsing:
![image](https://github.com/enso-org/enso/assets/4699705/74a555fc-b2b5-4801-888f-4c8f78d25a21)
